### ISSUE TITLE
INWX: Test PTR:Create_PTR_record failing

### ISSUE THREAD

**Issue by patschi** [StackExchange/dnscontrol#1948]
Currently it looks like one test with INWX is failing:

```
PS C:\dev\dnscontrol-src\integrationTest> go test -v -verbose -provider INWX -diff2
=== RUN   TestDNSProviders
=== RUN   TestDNSProviders/dnscontrol-inwx.com
[...]
--- FAIL: TestDNSProviders (151.26s)
    --- FAIL: TestDNSProviders/dnscontrol-inwx.com (150.96s)
[...]
        --- PASS: TestDNSProviders/dnscontrol-inwx.com/Post_cleanup:Empty#15 (0.54s)
        --- FAIL: TestDNSProviders/dnscontrol-inwx.com/18:PTR:Create_PTR_record (0.81s)
        --- PASS: TestDNSProviders/dnscontrol-inwx.com/Post_cleanup:Empty#16 (0.52s)
[...]
--- PASS: TestDualProviders (1.57s)
FAIL
exit status 1
FAIL    github.com/StackExchange/dnscontrol/v3/integrationTest  153.282s
```

**Comment by patschi** [comment]
I'm not able to reproduce this:
```
******************** Domain: domain.tld
1 correction
#1: CREATE PTR 1.domain.tld foo.example.com. ttl=43200
SUCCESS!
Done. 1 corrections.
```

Also visible in INWX web:
![image](https://user-images.githubusercontent.com/2029878/215280226-d49afb64-e2ec-41bd-81a7-63d58f4ffff6.png)

It's pretty much the same as:
https://github.com/StackExchange/dnscontrol/blob/a1366210526caaed5469d631b21afea5214d8dcc/integrationTest/integration_test.go#L951

The test command is running the commands against their sandbox environment, which is entirely separated from the production one. Either that's the cause for the failure, or something more related to the test itself.

**Comment by patschi** [comment]
Adding via WebUI on sandbox also works:
![image](https://user-images.githubusercontent.com/2029878/215282357-b6082a71-f2e0-47d4-9ffa-6cfe2b81ffef.png)

**Comment by tlimoncelli** [comment]
Hmm.. interesting!  Can you run the integration tests against a real domain?   You'll need a domain that isn't otherwise in use since the integration tests remove all records.

**Comment by patschi** [comment]
Yeah, I was planning to. Interestingly:

```
    integration_test.go:223: DELETE A rec0088.domain.tld 1.2.3.4 ttl=300
    integration_test.go:223: DELETE A rec0089.domain.tld 1.2.3.4 ttl=300
    integration_test.go:223: DELETE A rec0090.domain.tld 1.2.3.4 ttl=300
    integration_test.go:223: DELETE A rec0091.domain.tld 1.2.3.4 ttl=300
    integration_test.go:223: DELETE A rec0092.domain.tld 1.2.3.4 ttl=300
panic: test timed out after 10m0s

goroutine 3442 [running]:
testing.(*M).startAlarm.func1()
        /usr/lib/go-1.19/src/testing/testing.go:2036 +0x8e
created by time.goFunc
```

Apparently the production API is way slower than the sandbox and the test was timing out.

But the test even fails with the production instance:
```
$ go test -v -verbose -timeout 0 -provider INWX -start 19 -end 19
=== RUN   TestDNSProviders
=== RUN   TestDNSProviders/domain.tld
=== RUN   TestDNSProviders/domain.tld/Clean_Slate:Empty
=== RUN   TestDNSProviders/domain.tld/19:PTR:Create_PTR_record
    integration_test.go:223: CREATE PTR 4.domain.tld foo.com. ttl=300
    integration_test.go:242: Expected 0 corrections on second run, but found 1.
    integration_test.go:244: UNEXPECTED #0: MODIFY PTR 4.domain.tld: (foo.com ttl=300) -> (foo.com. ttl=300)
=== RUN   TestDNSProviders/domain.tld/Post_cleanup:Empty
    integration_test.go:223: DELETE PTR 4.domain.tld foo.com ttl=300
--- FAIL: TestDNSProviders (8.15s)
    --- FAIL: TestDNSProviders/domain.tld (7.76s)
        --- PASS: TestDNSProviders/domain.tld/Clean_Slate:Empty (0.30s)
        --- FAIL: TestDNSProviders/domain.tld/19:PTR:Create_PTR_record (5.08s)
        --- PASS: TestDNSProviders/domain.tld/Post_cleanup:Empty (2.37s)
=== RUN   TestDualProviders
    integration_test.go:332: Clearing everything
    integration_test.go:339: Adding nameservers from another provider
    integration_test.go:326: #1: CREATE NS domain.tld ns1.example.com. ttl=300
    integration_test.go:326: #2: CREATE NS domain.tld ns2.example.com. ttl=300
    integration_test.go:342: Running again to ensure stability
--- PASS: TestDualProviders (4.54s)
FAIL
exit status 1
FAIL    github.com/StackExchange/dnscontrol/v3/integrationTest  12.699s
```

**Comment by patschi** [comment]
I might have found the root cause:
```
    integration_test.go:242: Expected 0 corrections on second run, but found 1.
    integration_test.go:244: UNEXPECTED #0: MODIFY PTR 4.domain.tld: (foo.com ttl=300) -> (foo.com. ttl=300)
```

Testing the same with normal pushes:
```
******************** Domain: domain.tld
1 correction
#1: MODIFY PTR 4.domain.tld: (foo.com ttl=43200) -> (foo.com. ttl=43200)
SUCCESS!
Done. 1 corrections.
```

No matter how often I push, there's always the MODIFY correction for the PTR record.

**Comment by tlimoncelli** [comment]
Does it happen with and without `-diff2`?

**Comment by patschi** [comment]
Yes - just verified

**Comment by tlimoncelli** [comment]
Hmmm.... there is a period at the end of one string

**Comment by patschi** [comment]
Oh, indeed! Good catch. Looks like the value from INWX API is different than what is expected. I'll try to look into it.

**Comment by patschi** [comment]
Fix in PR #2430.
